ng build throws downlevelIteration error targeting es2015

# 🐞 Bug report

### Command (mark with an `x`)
<!-- Can you pin-point the command or commands that are effected by this bug? -->
<!-- ✍️edit: -->
```
- [ ] new
- [x] build
- [ ] serve
- [ ] test
- [ ] e2e
- [ ] generate
- [ ] add
- [ ] update
- [ ] lint
- [ ] xi18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc
```

### Is this a regression?

<!-- Did this behavior use to work in the previous version? -->
<!-- ✍️--> Yes, the previous version in which this bug was not present was: 7.3.*

### Description

Building with `ng build` while targeting `es2015` throws `downlevelIteration` error.

## 🔬 Minimal Reproduction

Create a new Angular application and use this code for `app.component.ts`

```
@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.css']
})
export class AppComponent implements OnInit {
  map = new Map<string, string>();

  ngOnInit(): void {
    this.map.set('key', 'value');

    for (const [k, v] of this.map) {
      console.log(k, v);
    }
  }
}
```

## 🔥 Exception or Error
<pre><code>
ERROR in src/app/app.component.ts(15,26): error TS2569: Type 'Map<string, string>' is not an array type or a string type. Use compiler option '--downlevelIteration' to allow iterating of iterators.
</code></pre>


## 🌍 Your Environment
<pre><code>
Angular CLI: 8.0.1
Node: 10.11.0
OS: win32 x64
Angular: 8.0.0
... animations, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.800.1
@angular-devkit/build-angular     0.800.1
@angular-devkit/build-optimizer   0.800.1
@angular-devkit/build-webpack     0.800.1
@angular-devkit/core              8.0.1
@angular-devkit/schematics        8.0.1
@angular/cli                      8.0.1
@ngtools/webpack                  8.0.1
@schematics/angular               8.0.1
@schematics/update                0.800.1
rxjs                              6.4.0
typescript                        3.4.5
webpack                           4.30.0
</code></pre>

`tsconfig.json`

<pre><code>
{
  "compileOnSave": false,
  "compilerOptions": {
    "baseUrl": "./",
    "outDir": "./dist/out-tsc",
    "sourceMap": true,
    "declaration": false,
    "module": "esnext",
    "moduleResolution": "node",
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "importHelpers": true,
    "target": "es2015",
    "typeRoots": [
      "node_modules/@types"
    ],
    "lib": [
      "es2018",
      "dom"
    ]
  }
}
</code></pre>

--- a/packages/schematics/angular/migrations/update-8/differential-loading.ts
+++ b/packages/schematics/angular/migrations/update-8/differential-loading.ts
@@ -131,7 +131,16 @@ function updateTsConfig(tree: Tree, tsConfigPath: string): void {
   if (isExtendedConfig) {
     removePropertyInAstObject(recorder, compilerOptions, 'target');
     removePropertyInAstObject(recorder, compilerOptions, 'module');
+    removePropertyInAstObject(recorder, compilerOptions, 'downlevelIteration');
   } else {
+    const downlevelIteration = findPropertyInAstObject(compilerOptions, 'downlevelIteration');
+    if (!downlevelIteration) {
+      insertPropertyInAstObjectInOrder(recorder, compilerOptions, 'downlevelIteration', true, 4);
+    } else if (!downlevelIteration.value) {
+      const { start, end } = downlevelIteration;
+      recorder.remove(start.offset, end.offset - start.offset);
+      recorder.insertLeft(start.offset, 'true');
+    }
     const scriptTarget = findPropertyInAstObject(compilerOptions, 'target');
     if (!scriptTarget) {
       insertPropertyInAstObjectInOrder(recorder, compilerOptions, 'target', 'es2015', 4);

--- a/packages/schematics/angular/migrations/update-8/differential-loading_spec.ts
+++ b/packages/schematics/angular/migrations/update-8/differential-loading_spec.ts
@@ -80,6 +80,29 @@ describe('Migration to version 8', () => {
       expect(module).toBe('esnext');
     });
 
+    it(`should create 'downlevelIteration' property when doesn't exists`, () => {
+      const tree2 = schematicRunner.runSchematic('migration-07', {}, tree.branch());
+      const compilerOptions = {
+        ...oldTsConfig.compilerOptions,
+      };
+
+      tree.overwrite(tsConfigPath, JSON.stringify({ compilerOptions }, null, 2));
+      const { downlevelIteration } = JSON.parse(tree2.readContent(tsConfigPath)).compilerOptions;
+      expect(downlevelIteration).toBe(true);
+    });
+
+    it(`should update 'downlevelIteration' to true when it's false`, () => {
+      const tree2 = schematicRunner.runSchematic('migration-07', {}, tree.branch());
+      const compilerOptions = {
+        ...oldTsConfig.compilerOptions,
+        downlevelIteration: false,
+      };
+
+      tree.overwrite(tsConfigPath, JSON.stringify({ compilerOptions }, null, 2));
+      const { downlevelIteration } = JSON.parse(tree2.readContent(tsConfigPath)).compilerOptions;
+      expect(downlevelIteration).toBe(true);
+    });
+
     it(`should create browserslist file if it doesn't exist`, () => {
       tree.delete('/browserslist');
       const tree2 = schematicRunner.runSchematic('migration-07', {}, tree.branch());

--- a/packages/schematics/angular/workspace/files/tsconfig.json.template
+++ b/packages/schematics/angular/workspace/files/tsconfig.json.template
@@ -5,9 +5,10 @@
     "outDir": "./dist/out-tsc",
     "sourceMap": true,
     "declaration": false,
+    "downlevelIteration": true,
+    "experimentalDecorators": true,
     "module": "esnext",
     "moduleResolution": "node",
-    "experimentalDecorators": true,
     "importHelpers": true,
     "target": "es2015",
     "typeRoots": [

